test/testhelpers.sh

```diff
@@ -427,8 +427,28 @@ comparison_to_operator() {
   fi
 }
 
+# Calculate the object ID from the string passed as the argument
 calc_oid() {
-  printf "$1" | shasum -a 256 | cut -f 1 -d " "
+  # Windows might be MSYS2 which does not have the shasum Perl wrapper
+  # script by default, so use sha256sum directly. MacOS on the other hand
+  # does not have sha256sum, so still use shasum otherwise.
+  if [ $IS_WINDOWS -eq 1 ]; then
+    printf "$1" | sha256sum
+  else
+    printf "$1" | shasum -a 256
+  fi | cut -f 1 -d " "
+}
+
+# Calculate the object ID from the file passed as the argument
+calc_oid_file() {
+  # Windows might be MSYS2 which does not have the shasum Perl wrapper
+  # script by default, so use sha256sum directly. MacOS on the other hand
+  # does not have sha256sum, so still use shasum otherwise.
+  if [ $IS_WINDOWS -eq 1 ]; then
+    sha256sum "$1"
+  else
+    shasum -a 256 "$1"
+  fi | cut -f 1 -d " "
```

Comment:
I guess there is no reasonable nice way to have the decision between `sha256sum` and `shasum` once in the code? Maybe a `sha256` function? I guess "eval" is too evil?!